decentraland/unity-renderer
# fix: optimizing PlayerName batching when it is turned off or turned on

## What does this PR change?

This PR removed gameObject.SetActive calls from PlayerName class in methods responsible for showing and hiding the player names because of various reasons. Instead it is using a list of canvas renderers and modifies their alpha which does not have such a huge performance overhead as gameObject SetActive

Closes #2713 

...

## How to test the changes?

Go with two peer avatar characters to the same area and check how the name of one character (over his head) is being shown and hidden from the perspective of other character during interactions like this:

1. Rotating camera when turning away from peer avatar character, then rotating back to him
2. Running away from the peer avatar character until his name disappears
3. Logging out on the peer avatar character side and seeing his name disappear together with him.
4. Teleporting peer avatar character and seeing his name disappear.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
